src/servers/mcp-server-base.ts

```diff
@@ -147,6 +148,41 @@ export abstract class BaseMCPServer extends Server {
         this.addTracker(mixpanel);
     }
 
+    protected async getTokenUrl(sessionId: string, generationNo: number) {
+        let tokenUrl = "";
+        // Generate token and store in KV store
+        if (this.ctx.env?.OAUTH_KV) {
```

Comment:
Lets put this in a `try ... catch`. The `putInKV` method should throw if `OAUTH_KV` is not available. And we should handle that condition in the `catch`